tests/test_optional_enhancements.py

```diff
@@ -0,0 +1,191 @@
+from app.models import Task, Goal
+import pytest
+
+
+def test_get_tasks_sorted_id_asc(client, three_tasks):
+    # Act
+    response = client.get("/tasks?sort=asc&by=task_id")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert len(response_body) == 3
+    assert response_body == [
+        {
+            "description": "",
+            "id": 1,
+            "is_complete": False,
+            "title": "Water the garden 🌷",
+        },
+        {
+            "description": "",
+            "id": 2,
+            "is_complete": False,
+            "title": "Answer forgotten email 📧",
+        },
+        {
+            "description": "",
+            "id": 3,
+            "is_complete": False,
+            "title": "Pay my outstanding tickets 😭",
+        },
+    ]
+
+
+def test_get_tasks_sorted_id_desc(client, three_tasks):
+    # Act
+    response = client.get("/tasks?sort=desc&by=task_id")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert len(response_body) == 3
+    assert response_body == [
+        {
+            "id": 3,
+            "title": "Pay my outstanding tickets 😭",
+            "description": "",
+            "is_complete": False,
+        },
+        {
+            "id": 2,
+            "title": "Answer forgotten email 📧",
+            "description": "",
+            "is_complete": False,
+        },
+        {
+            "id": 1,
+            "title": "Water the garden 🌷",
+            "description": "",
+            "is_complete": False,
+        },
+    ]
+
+
+def test_get_tasks_invalid_sort(client, three_tasks):
+    # Act
+    response = client.get("/tasks?sort=asdf")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
```

Comment:
I guess this is fine given how you've implemented this, but usually I would expect we would return a 400 here so the user knows they asked to sort on an invalid key to sort on.

Reply:
more copy paste mistakes lol